Tests/LegacyUnit/BagBuilder/AbstractTest.php

```diff
@@ -520,6 +520,73 @@ public function testWhereClauseCanSelectPids()
         );
     }
 
+    public function testWhereClausePartGetKeyMustNotBeEmpty()
+    {
+        $this->expectException(
+            \InvalidArgumentException::class
+        );
+        $this->expectExceptionMessage(
+            'The parameter $key must not be empty.'
+        );
+
+        $this->subject->getWhereClausePart('');
+    }
+
+    public function testWhereClausePartSetKeyMustNotBeEmpty()
+    {
+        $this->expectException(
+            \InvalidArgumentException::class
+        );
+        $this->expectExceptionMessage(
+            'The parameter $key must not be empty.'
+        );
+
+        $this->subject->setWhereClausePart('', '');
+    }
+
+    public function testWhereClausePartInitiallyIsEmpty()
+    {
+        self::assertEquals(
+            '',
+            $this->subject->getWhereClausePart('testpart')
+        );
+    }
+
+    public function testWhereClausePartCanBeSetAndCanBeRetrieved()
+    {
+        $this->subject->setWhereClausePart('testpart', 'testpart IN (1,2,3)');
```

Comment:
I propose we also have a test that tests that the set part is used in the query.

Reply:
`whereClausePartCanBeSetAndGetsAddedToWhereClause()` is checking for the part being in the string returned by `getWhereClause()`. Is that enough?

Reply:
Now that I think about this, yes, it is. Thanks!